executor/set_config.go

```diff
@@ -0,0 +1,167 @@
+		if !isValidInstance(s.p.Instance) {
+			return errors.Errorf("invalid instance %v", s.p.Instance)
+		}
+	}
+	s.p.Name = strings.ToLower(s.p.Name)
+
+	val, isNull, err := s.p.Value.EvalString(s.ctx, chunk.Row{})
+	if err != nil {
+		return err
+	}
+	if isNull {
+		return errors.Errorf("can't set config to null")
+	}
+	s.v = val
+	return nil
+}
+
+// TestSetConfigServerInfoKey is used as the key to store 'TestSetConfigServerInfoFunc' in the context.
+var TestSetConfigServerInfoKey stringutil.StringerStr = "TestSetConfigServerInfoKey"
+
+// TestSetConfigHTTPHandlerKey is used as the key to store 'TestSetConfigDoRequestFunc' in the context.
+var TestSetConfigHTTPHandlerKey stringutil.StringerStr = "TestSetConfigHTTPHandlerKey"
+
+// Next implements the Executor Next interface.
+func (s *SetConfigExec) Next(ctx context.Context, req *chunk.Chunk) error {
+	req.Reset()
+	getServerFunc := infoschema.GetClusterServerInfo
+	if v := s.ctx.Value(TestSetConfigServerInfoKey); v != nil {
+		getServerFunc = v.(func(sessionctx.Context) ([]infoschema.ServerInfo, error))
+	}
+
+	serversInfo, err := getServerFunc(s.ctx)
+	if err != nil {
+		return err
+	}
+	nodeTypes := set.NewStringSet()
+	nodeAddrs := set.NewStringSet()
+	if s.p.Type != "" {
+		nodeTypes.Insert(s.p.Type)
+	}
+	if s.p.Instance != "" {
+		nodeAddrs.Insert(s.p.Instance)
+	}
+	serversInfo = filterClusterServerInfo(serversInfo, nodeTypes, nodeAddrs)
+
+	for _, serverInfo := range serversInfo {
+		var url string
+		switch serverInfo.ServerType {
+		case "pd":
+			url = fmt.Sprintf("%s://%s%s", util.InternalHTTPSchema(), serverInfo.StatusAddr, pdapi.Config)
+		case "tikv":
+			url = fmt.Sprintf("%s://%s/config", util.InternalHTTPSchema(), serverInfo.StatusAddr)
+		default:
+			continue
+		}
+		if err := s.doRequest(url); err != nil {
+			s.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
+		}
+	}
+	return nil
+}
+
+func (s *SetConfigExec) doRequest(url string) error {
+	body := bytes.NewBufferString(fmt.Sprintf("{'%s':'%s'}", s.p.Name, s.v))
+	req, err := http.NewRequest(http.MethodPost, url, body)
+	if err != nil {
+		return err
+	}
+	var httpHandler func(req *http.Request) (*http.Response, error)
+	if v := s.ctx.Value(TestSetConfigHTTPHandlerKey); v != nil {
+		httpHandler = v.(func(*http.Request) (*http.Response, error))
+	} else {
+		httpHandler = util.InternalHTTPClient().Do
+	}
+	resp, err := httpHandler(req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode == http.StatusOK {
+		return nil
+	} else if resp.StatusCode >= 400 && resp.StatusCode < 600 {
+		message, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		return errors.Errorf("bad request to %s: %s", url, message)
+	}
+	return errors.Errorf("request %s failed: %s", url, resp.Status)
+}
+
+func isValidInstance(instance string) bool {
+	var ip, port string
+	for i := len(instance) - 1; i >= 0; i-- {
+		if instance[i] == ':' {
+			ip = instance[:i]
+			port = instance[i+1:]
+		}
```

Comment:
Better to use `net.SplitHostPort` to handle IPv6 IP.

Reply:
Addressed. PTAL.